### PR TITLE
put something useful in boards/README.md

### PR DESCRIPTION
**Pull request by fhars**
### Pull Request Overview

This addresses one of the issues mentioned in #1402

### Testing Strategy

This pull request was tested by clicking on all the links to make sure they point to the right board specific README.

### Documentation Updated

- [X] Updated the relevant files in `/docs`, or no updates are required.

### Formatting

- [X] Ran `make formatall`.

### CODE REVIEW

File: boards/README.md [hunk 1]
```diff
@@ -3,3 +3,16 @@ Platforms Supported by Tock
...
+|---------------------------------------------------|---------------|------------|------------|----------------|
+| [Hail](hail/README.md)                            | ARM Cortex-M4 | SAM4LC8BA  | Bootloader | tockloader     |
+| [Imix](imix/README.md)                            | ARM Cortex-M4 | SAM4LC8CA  | Bootloader | tockloader     |
+| [Nordic nRF52-DK](nordic/nrf52dk/README.md)       | ARM Cortex-M4 | nRF52832   | jLink      | tockloader     |
+| [Nordic nRF52840-DK](nordic/nrf52840dk/README.md) | ARM Cortex-M4 | nRF52840   | jLink      | tockloader     |
+| [ACD52832](acd52832/README.md)                    | ARM Cortex-M4 | nRF52832   | jLink      | tockloader     |
+| [TI LAUNCHXL-CC26x2](launchxl/README.md)          | ARM Cortex-M4 | CC2652R    | openocd    | tockloader     |
+| [ST Nucleo F446RE](nucleo_f446re/README.md)       | ARM Cortex-M4 | STM32F446  | openocd    | custom         |
+| [ST Nucleo F429ZI](nucleo_f429zi/README.md)       | ARM Cortex-M4 | STM32F429  | openocd    | custom         |
+| [SiFive HiFive1](hifive1/README.md)               | RISC-V        | FE310-G000 | openocd    | not supported  |
```
Comment (alistair23) [inline boards/README.md]: App deployment is supported.

Comment (fhars) [reply boards/README.md]: I just copied the information from the README:

> The RISC-V core on this board only supports Machine mode. That means that this
board does not support running userland applications.

Comment (alistair23) [reply boards/README.md]: Ah, that is out of date then. RISC-V does support userspace apps now

Comment (alistair23) [reply boards/README.md]: I was planning on updating the README, so I'll fix it in the board README when I do that.

Comment (alistair23) [reply boards/README.md]: This was never updated.

**Pull request merged**